asf_core_data/pipeline/mcs/process/process_historical_mcs_installers.py

```diff
@@ -0,0 +1,926 @@
+
+2) To process another version of the raw historical installers data
+(where YYYYMMDD corresponds to the date the data was shared by MCS in the data dumps Google Drive folder, e.g. 20230207):
+`export COMPANIES_HOUSE_API_KEY="YOUR_API_KEY"`
+`python3 process_historical_mcs_installers.py -raw_historical_installers_filename "raw_historical_mcs_installers_YYYYMMDD.xlsx" -raw_historical_installations_filename "raw_historical_mcs_installations_YYYYMMDD.xlsx"`
+OR
+`python3 process_historical_mcs_installers.py -api_key "YOUR_API_KEY" -raw_historical_installers_filename "raw_historical_mcs_installers_YYYYMMDD.xlsx" -raw_historical_installations_filename "raw_historical_mcs_installations_YYYYMMDD.xlsx"`
+
+Companies House API additional info:
+- https://developer-specs.company-information.service.gov.uk/companies-house-public-data-api/reference/search/search-companies
+- https://developer-specs.company-information.service.gov.uk/companies-house-public-data-api/resources/companysearch?v=latest
+
+"""
+
+import pandas as pd
+import numpy as np
+import requests
+import random
+import time
+import os
+from datetime import datetime
+from argparse import ArgumentParser
+from asf_core_data.config import base_config
+from asf_core_data.getters.data_getters import s3, load_s3_data, save_to_s3
+from asf_core_data.getters.mcs_getters.get_mcs_installers import (
+    get_raw_historical_mcs_installers,
+)
+from asf_core_data.getters.mcs_getters.get_mcs_installations import (
+    get_raw_historical_installations_data,
+)
+
+
+def basic_preprocessing_of_installations(raw_historical_installations: pd.DataFrame):
+    """
+    Applies basic preprocessing to raw installations data:
+    - Renaming columns (lower case, spaces to underscore, etc.);
+    - Extracting certification body and certification number info from original var.
+    Args:
+        raw_historical_installations: raw historical installations table
+    """
+
+    raw_historical_installations.columns = rename_columns(
+        raw_historical_installations.columns
+    )
+
+    raw_historical_installations["certification_body"] = (
+        raw_historical_installations["installation_company_mcs_number"]
+        .str.split(" ")
+        .str[0]
+    )
+
+    raw_historical_installations["installation_company_mcs_number"] = (
+        raw_historical_installations["installation_company_mcs_number"]
+        .str.split(" ")
+        .str[1]
+        .astype(int)
+    )
+
+
+def rename_columns(cols: list) -> list:
+    """
+    Renames a list of strings, e.g. a list of column names, by:
+    - applying lower case;
+    - replacing "heat pump" by "hp";
+    - replacing " " and "/" by "_";
+    - correcting the word address.
+
+    Args:
+        cols: original column names
+    Returns:
+        Renamed column names
+    """
+    cols = [c.lower() for c in cols]
+    changes_dict = {"heat pump": "hp", " ": "_", "/": "_", "adddress": "address"}
```

Comment:
Didn't check whether this ends up being a problem in practice, but since dicts are not ordered, I worry that the following could happen: In the string 'heat pump', the white space is replaced with '_' and so 'heat_pump' will not be shortened to 'hp'. Easiest fix is adding an additonal replacement entry:

```suggestion
    changes_dict = {"heat pump": "hp", "heat_pump":"hp", " ": "_", "/": "_", "adddress": "address"}
```

Reply:
I don't think this would ever be a problem in practice, but I will add a test to check this in the test script :)